scipy/meson.build

```diff
@@ -67,13 +67,21 @@ lapack = dependency('openblas')
 #       _build_utils/_fortran.py)
 g77_abi_wrappers = files('_build_utils' / 'src' / 'wrap_dummy_g77_abi.f')
 
+config = files('../tools/config_utils.py')
+version = files('../tools/version_utils.py')
+generate_config =  run_command(py3, config, '__config__.py')
+generate_version =  run_command(py3, version)
```

Comment:
There's a couple of things to look at here:

1. Is `run_command` the correct thing to use? I think yes, because it should always be run (unlike other code generation steps with `custom_target`).
2. Should it use `check: true` to catch failures? Probably yes, seems healthy to catch any failures.
3. Do the generated files end up in the right place? I think no; since they are generated they should be in `build/scipy/` rather than `scipy/`. Also keeping in mind the `run_command` docs, _"The command is run from an unspecified directory ...."_. So it looks like this should be using `MESON_BUILD_ROOT`. On the other hand, `run_command` executes at configure time rather than at build time. So maybe it's okay.
4. Does it actually work? `__config__.py` looks fine to me, however `version.py` looks like:
```
# THIS FILE IS GENERATED FROM SCIPY SETUP.PY
short_version = '1.8.0'
version = '1.8.0'
full_version = '1.8.0.dev0+Unknown.Unknown'
git_revision = 'Unknown'
commit_count = 'Unknown'
release = False

if not release:
    version = full_version
```
for me. What does it look like for you @HarshCasper?

Reply:
Okay I've read the docs again and convinced myself that `run_command` is only for build configuration and not for code generation. I think this should be changed to `custom_target`, with `build_always_stale: true`, `build_by_default: true`, and `install: true` (see https://mesonbuild.com/Reference-manual.html#custom_target). And it should go into `build/scipy`.

Reply:
Hi @rgommers 

Regarding point 4, this is what it looks like for me: 


```py
# THIS FILE IS GENERATED FROM SCIPY SETUP.PY
short_version = '1.8.0'
version = '1.8.0'
full_version = '1.8.0.dev0+Unknown.Unknown'
git_revision = 'Unknown'
commit_count = 'Unknown'
release = False

if not release:
    version = full_version
```

Reply:
Yes, that's what I see as well. The `Unknown`'s mean that after the changes in this PR, the code fails to pick up the git revision info. It should look like:
```
full_version = '1.8.0.dev0+1521.baabfad'
git_revision = 'baabfad'
commit_count = '1521'
```

Reply:
I'm still seeing the `Unknown`s with commit b7aa96131b.

Reply:
It's not completely trivial to figure out why - probably a combination of the build dir the code is executed from plus the `subprocess.Popen` call. It works fine if you just call the script, just not via `ninja -C build`.

Reply:
If it's not easy to figure out tomorrow, I'm fine with opening a separate issue for it and merging this PR. The other comments should be easy to address.